tests/test_foreman.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from urllib.parse import urlparse
+
+
+def test_foreman_version(api, variables):
+    if 'foreman_version' not in variables:
+        pytest.skip("'foreman_version' is not set")
+
+    status = api.resource('home').call('status')
+
+    assert status['version'] == variables['foreman_version']
+
+
+def test_ping(api):
+    ping = api.resource('ping').call('ping')['results']
+
+    assert ping['foreman']['database']['active']
+
+    if 'katello' in ping:
+      assert ping['katello']['status'] == 'ok'
+
+
+def test_check_smart_proxy_registered(api, base_url):
+    hostname = urlparse(base_url).hostname
+    assert api.resource('smart_proxies').call('show', {'id': hostname})
```

Comment:
This asserts that there is a Smart Proxy on the same host as Foreman. However, that is not always true. For example, on my own Foreman server there is just Foreman, no Smart Proxy. That's on a separate server. We should not bake in this assumption that isn't true.

I think expected proxies as a variable is a better check. That also allows asserting that there is an external Smart Proxy like we have in the Katello test setup.

Reply:
I think I can make that improvement.